Dm 2854

JoshingYou1 opened the pull request:
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2854
https://agile6.atlassian.net/browse/DM-2862

## Description - what does this code do?
Adds content to about page (desktop view only) and addresses filtering feature for maturity level links.

## Testing done - how did you test it/steps on how can another person can test it 
1. Visit '/about'
2. Click on the `Diffusion of Excellence` link in the intro section and make sure you're taken to `va.gov` in a new window.
3. Go back and open the first accordion, click on the `HAPPEN` link, and make sure you're taken to `Project HAPPEN`'s show page.
4. Click on the `How does diffusion work?` accordion and then choose one of the maturity level links.
5. Make sure you're taken to the search page with ONLY practices that have the same `maturity_level` as the one you chose (if need be, update a few practices with a maturity level and then repeat these steps).
6. Click on the `How do I use the Diffusion Marketplace?` accordion, click on the `Start a search` link, and make sure you're taken to the search page with a clean slate (no filters, input text, etc.).
7. Scroll down to the `Contact us` section.
8. Click on the `marketplace@va.gov` link and make sure your email app of choice opens with the content `To: marketplace@va.gov`, and that all of the other fields are blank.
9. Now fill out the form (every field is required, so you can test that as well) and then click `Send message`.
10. Make sure you get a new mailer window with the correct data (From: `The email you used`, Subject: `The subject you used`, and the message you used should be in the body of the email).
11. Also make sure you get a notification at the top of the page stating you successfully sent a message.
12. **BONUS**: Per design, the alert messages can now be removed by the user, so click the `X` on the right side of the notification and make sure it gets removed from the DOM.

## Screenshots, Gifs, Videos from application (if applicable)
![](https://media.giphy.com/media/MNHXaG63FQ8y7WlYOD/giphy.gif?cid=790b76117b98378da18f473e074bd7e88e2e793d2616c430&rid=giphy.gif&ct=g)
![](https://media.giphy.com/media/LHfvdNx0I2sLZ2kRsN/giphy.gif?cid=790b7611d0f78d1ef58f7ab4e28819d64f10a8c2682d74b6&rid=giphy.gif&ct=g)
![](https://media.giphy.com/media/79V3B1NDwCxj9vwVsw/giphy.gif?cid=790b76115919c276dd880e0a68e0d6aba80a40680449560f&rid=giphy.gif&ct=g)


## Link to mock-ups/mock ups (image file if you have it) (if applicable)
https://www.figma.com/file/xjaSwxytWsGGGCcc9f9Yfa/Handoff?node-id=293%3A117

## Acceptance criteria
- Build intro section
- Build About page content
- Build FAQ section with accordions
- Build Contact Us mailer section

## Definition of done
- [X] Unit tests written (if applicable)
- [X] e2e/accessibility tests written (if applicable)
- [X] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

bradjohnson92008 reviewed (changes requested):
Great Job Josh... only issue I had was with step #5... only "emerging" categories were displaying for me...  both "replicating" and "scaling" did not display.  I switched the two emerging innovations (haha) to replicating just to make sure it wasn't some other flag effecting the search results - but still the innovations didn't display for both 'replicating' and 'scaling'.  To test I just updated the practice.maturity_level field to 0, 1, or 2..

emyl3 reviewed (changes requested):
locally for me going to filter by scaling and replicating did not work. I also had a couple questions/comments for you! Let me know if you have any questions :) Thanks, Josh!

JoshingYou1 commented on app/views/practices/show/desktop_partials/_maturity_level_banner.html.erb:
```diff
@@ -6,7 +6,7 @@
     <p class="usa-alert__text margin-bottom-0">
       <%= maturity_map[:description] %>
-      <a href="/search?query=<%= maturity %>" class="usa-link">See more <%= maturity_map[:link_text] %> practices.</a>
+      <a href="/search#filter_by=<%= maturity %>" class="usa-link">See more <%= maturity_map[:link_text] %> practices.</a>
```
We don't want to pass query params, because we are doing a filtered search. We want to separate the two use cases in which, A. the user actually types in replicate in the search input and B. The user clicks on a maturity level link. Hope this makes sense.

emyl3 replied:
I guess I am asking does `/search?filter_by` not work since I have only seen the `#` symbol in urls help to navigate to a certain section of the page if that makes sense 😅

JoshingYou1 replied:
Okay, I understand. Yes, I believe it would work. If you think this is the better way to go, I will make that change. Thanks Elisa!

JoshingYou1 commented on app/views/about/index.html.erb:
```diff
@@ -1,5 +1,125 @@
+        <h1 class="usa-prose-h1 margin-bottom-4 margin-top-0">About us</h1>
+        <p class="usa-prose-body">
+          The Diffusion Marketplace is a discovery and collaboration tool that curates promising clinical, operational, and strategic practices from VA. We are managed by the
+          <a href="https://www.va.gov/INNOVATIONECOSYSTEM/views/who-we-are/diffusion-of-excellence.html" class="dm-primary-link" target="_blank">Diffusion of Excellence</a>, an organization within
```
I'm not sure. In the homepage branch, I created a new `dm-primary-link` that we can use if we need the bold stuff.

emyl3 replied:
if it's just this page I would use the `usa-link text-bold`. We'll wait and see what Becca says. I also asked if we should make this a `usa-link usa-link--external text-bold` link since it is technically outside of Diffusion Marketplace.

JoshingYou1 replied:
Okay, I refactored them and removed the `dm-primary-link` style

JoshingYou1 commented on app/views/shared/_messages.erb:
```diff
@@ -15,7 +19,10 @@
-          <div class="usa-alert__heading <%= small_text ? 'font-sans-sm margin-bottom-0' : '' %>" role="alert"><%= value %></div>
+          <div class="usa-alert__heading <%= small_text ? 'font-sans-sm margin-bottom-0' : '' %>" role="alert">
+            <%= value %>
+            <i id="close-alert-message" class="fas fa-times position-absolute right-3 cursor-pointer"></i>
```
Not sure. I'll have to ask Becca. I would assume she wouldn't want it to just be unique to the about page, but I'll ask her and let you know.

emyl3 replied:
Thank you! :D

JoshingYou1 replied:
She said it should be applied to any message on any page :)

JoshingYou1 commented:
> locally for me going to filter by scaling and replicating did not work. I also had a couple questions/comments for you! Let me know if you have any questions :) Thanks, Josh!

This should be resolved now. Thank you for catching this!

JoshingYou1 commented:
> Great Job Josh... only issue I had was with step #5... only "emerging" categories were displaying for me... both "replicating" and "scaling" did not display. I switched the two emerging innovations (haha) to replicating just to make sure it wasn't some other flag effecting the search results - but still the innovations didn't display for both 'replicating' and 'scaling'. To test I just updated the practice.maturity_level field to 0, 1, or 2..

This should be resolved now. Thank you Brad!

emyl3 reviewed (commented):
The replicate and scale filters are working for me! Great catch, Josh. Let's see what Becca has to say about the other comments :)

bradjohnson92008 reviewed (approved):
Retested the search features for maturity levels... and all is working as expected.  Great work Josh!

emyl3 reviewed (approved):
looks good to me, Josh!!
